linera-views/benches/stores.rs

```diff
@@ -0,0 +1,547 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{
+    fmt::Debug,
+    time::{Duration, Instant},
+};
+
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
+#[cfg(with_dynamodb)]
+use linera_views::dynamo_db::{create_dynamo_db_test_store, DynamoDbStore};
+#[cfg(with_rocksdb)]
+use linera_views::rocks_db::{create_rocks_db_test_store, RocksDbStore};
+#[cfg(with_scylladb)]
+use linera_views::scylla_db::{create_scylla_db_test_store, ScyllaDbStore};
+use linera_views::{
+    batch::Batch,
+    common::LocalKeyValueStore,
+    memory::{create_memory_store, MemoryStore},
+    test_utils::{get_random_key_values2, prefix_expansion},
+};
+use tokio::runtime::Runtime;
+
+pub async fn performance_contains_key<S: LocalKeyValueStore>(store: S, iterations: u64) -> Duration
+where
+    S::Error: Debug,
+{
+    let prefix = vec![0];
+    let num_entries = 100;
+    let len_value = 10000;
+
+    let mut total_time = Duration::ZERO;
+    for _ in 0..iterations {
+        let key_values = prefix_expansion(&prefix, get_random_key_values2(num_entries, len_value));
+        let mut batch = Batch::new();
+        for key_value in &key_values {
+            batch.put_key_value_bytes(key_value.0.clone(), key_value.1.clone());
+        }
+        store.write_batch(batch, &[]).await.unwrap();
+
+        let measurement = Instant::now();
+        for key_value in &key_values {
+            black_box(store.contains_key(&key_value.0).await.unwrap());
+        }
+        total_time += measurement.elapsed();
+
+        let mut batch = Batch::new();
+        batch.delete_key_prefix(prefix.clone());
+        store.write_batch(batch, &[]).await.unwrap();
+    }
+
+    total_time
```

Comment:
Optional: all of the `performance_*` functions follow a similar template. I'm wondering if it could be extracted into a separate helper function in the file :thinking:

Reply:
I extracted the commonalities between the tests but I did not go into a templatized version of the code that may have been overengineered.